libraries/botbuilder-applicationinsights/src/telemetryInitializerMiddleware.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+
+import { BotTelemetryClient, NullTelemetryClient, TelemetryLoggerMiddleware } from 'botbuilder-core';
+import { Middleware } from 'botbuilder-core';
+import { TurnContext } from 'botbuilder-core';
+import { TelemetryClient } from 'applicationinsights';
+import { CorrelationContext } from 'applicationinsights/out/AutoCollection/CorrelationContextManager';
+import * as appInsights from 'applicationinsights';
+
+/**
+ * Middleware for storing the incoming activity to be made available to Application Insights and optionally run the TelemetryLoggerMiddleware.
+ * Uses the botTelemetryClient interface.
+ */
+export class TelemetryInitializerMiddleware implements Middleware {
+
+    // tslint:disable:variable-name
+    private readonly _logActivityTelemetry: boolean;
+    private readonly _telemetryLoggerMiddleware: TelemetryLoggerMiddleware;
+    // tslint:enable:variable-name
+
+    /**
+     * Initializes a new instance of the TelemetryInitializerMiddleware class.
+     * @param telemetryLoggerMiddleware The TelemetryLoggerMiddleware used for logging activity telemetry.
+     * * @param logActivityTelemetry (Optional) Enable/Disable logging of activity telemetry.
+     */
+    constructor(telemetryLoggerMiddleware: TelemetryLoggerMiddleware, logActivityTelemetry: boolean = false) {
+        this._telemetryLoggerMiddleware = telemetryLoggerMiddleware;
+        this._logActivityTelemetry = logActivityTelemetry;
+    }
+
+    /**
+     * Gets a value indicating whether determines whether to log personal information that came from the user.
```

Comment:
>whether to log personal information [](start = 50, length = 35)

Hmm - might want to update..

Reply:
@daveta Thanks.  Will update.  FYI, the currently checked in unit test is not passing right now.  I have just got it running and will be checking in tests shortly.

Reply:
@daveta Comment updated.  Also additional unit tests now in place and passing.